src/miner.cpp

```diff
@@ -529,8 +529,10 @@ CBlockTemplate* BlockAssembler::CreateNewBlock(
         if (nHeight >= chainparams.GetConsensus().nZnodePaymentsStartBlock) {
             const Consensus::Params &params = chainparams.GetConsensus();
             CAmount znodePayment = GetZnodePayment(chainparams.GetConsensus(), nHeight > 0 && nBlockTime >= params.nMTPSwitchTime);
-            coinbaseTx.vout[0].nValue -= znodePayment;
-            FillBlockPayments(coinbaseTx, nHeight, znodePayment, pblock->txoutZnode, pblock->voutSuperblock);
+            if(mnodeman.CountEnabled() > 0){
```

Comment:
I'm not familiar with znodes. Can it happen, that mnodeman.CountEnabled() > 0, but they did not agree on the znode winner, and we still want to pay it to miner?

Reply:
it checked either we have znodes running on network or not